src/main/groovy/com/athaydes/automaton/SwingUtil.groovy

```diff
@@ -137,12 +137,23 @@ class SwingUtil {
 		}
 		for ( row in rows ) {
 			for ( col in cols ) {
-				if ( action( table.model.getValueAt( row, col ), row, col ) ) return true
+				if ( action( getRenderedTableValue(table, row, col ), row, col ) ) return true
 			}
 		}
 		return false
 	}
 
+	private static Object getRenderedTableValue(JTable table, row, col) {
+		def value = table.model.getValueAt(row, col)
+		def rendererComp = table.getCellRenderer(row, col)
+				.getTableCellRendererComponent(table, value, false, false, row, col)
+		if (rendererComp instanceof JLabel) {
+			(rendererComp as JLabel).getText()
```

Comment:
This only works for JLabel, which is a pretty big restriction... the approach I use, normally, is  to call `SwingUitl.callMethodIfExists(rendererComp, 'getText')` which works on a number of different types.